src/protocol/attribution/credit_capping.rs

```diff
@@ -40,6 +40,12 @@ where
     }
     let input_len = input.len();
 
+    if (u128::from(cap) * 2) >= F::PRIME.into() {
+        return Err(crate::error::Error::InvalidQueryParameter(format!(
+            "The cap {cap} must be less than 1/2 of the prime number to avoid overflow.",
```

Comment:
Technically, this does not prevent overflow. We can still have it, but it becomes detectable. Perhaps:

```
The cap {cap} must be less than 1/2 of the prime modulus to make overflow detectable, and propagable.
```